modules-js/react-fleet/src/elements/inputs/FileInput.tsx

```diff
@@ -0,0 +1,203 @@
+}
+
+interface fileSize {
+  quantity: string | number;
+  unit: string;
+}
+
+export default class FileInput extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      selectedFile: {},
+      isFocused: false
+    }
+  }
+
+  private inputRef = React.createRef<HTMLInputElement>();
+
+  /**
+   * Only accept the file if it doesn’t exceed the specified limit
+   */
+  private checkSize = (file: any) => {
+    const { quantity, unit } = this.props.sizeLimit;
+    const result = formatBytes(file.size);
+
+    if (result.unit === unit && +quantity >= +quantity) {
+      // todo: should this be a modal/notification component?
+      return alert(`File size limit of ${quantity}${unit} exceeded.\n\nYour file is ${result.quantity + result.unit}. Please select a different file.`);
+    }
+
+    this.setState({ selectedFile: file });
+  };
+
+  /**
+   * Ensure file is cleared in the state, as well as in the input element itself
+   */
+  private clearFile = () => {
+    // @ts-ignore
+    this.setState({ selectedFile: {} }, () => this.inputRef.current.value = '');
+  };
+
+  private setFocus = (isFocused: boolean) => {
+    this.setState({ isFocused });
+  };
+
+  private handleChange = () => {
+    // @ts-ignore
+    if (!this.inputRef.current.files.length) {
+      this.clearFile();
+
+    } else {
+      // @ts-ignore
+      this.checkSize(this.inputRef.current.files[0]);
+    }
+  };
+
+  private handleBlur = () => {
+    // todo: parent passes down handler method to collect the file?
+    // todo: or formik will manage?
+    // this.props.parentFileHandler(this.state.selectedFile);
+
+    this.setFocus(false);
+  };
+
+  render() {
+    return (
+      <div className={`txt m-b300 ${INPUT_CONTAINER_STYLE}`}>
+        <input
+          className={VISUALLYHIDDEN}
+          ref={this.inputRef}
+          type="file"
+          accept={this.props.fileTypes}
+          id={`FileInput-${this.props.name}`}
+          name={this.props.name}
+          onChange={this.handleChange}
+          onBlur={this.handleBlur}
+          onFocus={() => this.setFocus(checkFocus(this.props.name))}
+        />
+
+        <label
+          htmlFor={`FileInput-${this.props.name}`}
+          className={`btn btn--sm btn--100 ${this.state.isFocused && LABEL_FOCUSED_STYLE}`}
+          style={{ whiteSpace: 'nowrap' }}
+        >
+          Choose {this.props.title} file
+        </label>
+
+        <div className={FILE_PREVIEW_STYLE}>
+          <span>{this.state.selectedFile.name || DEFAULT_PREVIEW_TEXT}</span>
+        </div>
+
+        {this.state.selectedFile.name && (
+          <CloseButton
+            className={DELETE_BUTTON_STYLE}
+            size="1.8em"
+            title="Remove file"
+            handleClick={this.clearFile}
+          />
+        )}
+      </div>
+    );
+  }
+}
+
+
+function checkFocus(fieldName: string): boolean {
+  return document.activeElement.getAttribute('name') === fieldName;
```

Comment:
Can you comment about why this check is needed, given that we're already in an `onFocus` handler?

Reply:
Ah, thanks for the catch - this should have been refactored out a while ago! Next PR will reflect.